tools/github-actions/cascading/src/cascading.ts

```diff
@@ -0,0 +1,284 @@
+            .reduce((result: { packageChanges: { file: string; oldPackages: string[]; newPackages: string[] }[]; yarnLockFile?: string }, fileDiff) => {
+              // Only do the processing on the interesting parts
+              if (fileDiff.length > 10) {
+                const filenameResults = fileDiff.match(/(\S+\.\w+)/);
+                this.options.logger.info(`File being processed: ${fileDiff}`);
+                const filename = filenameResults && filenameResults[0];
+                if (filename) {
+                  if (filename.includes('package.json')) {
+                    result.packageChanges.push({file: filename, ...extractPackages(fileDiff)});
+                  } else {
+                    result.yarnLockFile = filename;
+                  }
+                }
+              }
+              return result;
+            }, {packageChanges: []});
+
+          const nonIgnoringChanges = changes.packageChanges.reduce((remaining: { file: string; addedPackages: string[]; commonPackages: string[]; removedPackages: string[] }[], fileWithPackages) => {
+            if (fileWithPackages.oldPackages.length || fileWithPackages.newPackages.length) {
+              const removedPackages: string[] = [];
+              const commonPackages: string[] = [];
+              fileWithPackages.oldPackages.forEach((oldPackage) => {
+                if (!fileWithPackages.newPackages.includes(oldPackage)) {
+                  removedPackages.push(oldPackage);
+                } else {
+                  if (!this.options.conflictsIgnoredPackages.includes(oldPackage)) {
+                    // We keep it only if it is not ignored
+                    commonPackages.push(oldPackage);
+                  }
+                }
+              });
+              const addedPackages = fileWithPackages.newPackages.filter((newPackage) => !fileWithPackages.oldPackages.includes(newPackage));
+              if (addedPackages.length || commonPackages.length || removedPackages.length) {
+                remaining.push({
+                  file: fileWithPackages.file,
+                  addedPackages,
+                  commonPackages,
+                  removedPackages
+                });
+              }
+            }
+            return remaining;
+          }, []);
+
+          if (nonIgnoringChanges.length) {
+            this.options.logger.info('Conflicts in package.json cannot be ignored');
+            nonIgnoringChanges.forEach((fileChange) => {
+              this.options.logger.info(JSON.stringify(fileChange, null, 2));
+            });
+            await this.createFallbackPullRequest(branchToCascade);
+            return;
+          } else {
+            this.options.logger.info('Conflicts can be ignored, discarding those changes');
+            const listOfDiscardingFiles = [...changes.packageChanges.map((file) => file.file), changes.yarnLockFile];
+            for (const file of listOfDiscardingFiles) {
+              if (file) {
```

Comment:
this check is only necessary because of line 174, it could be done so we only check the presence of `changes.yarnLockFile`